File: mass/commands.cpp
// minstd_rand
#include <random>
#include <iostream>
#include <vector>
#include <cctype>
#include <algorithm>

using namespace std;

// Задание 1 
vector<int> GenerateArray1(int n) {
    vector<int> arr1;
    minstd_rand gen(random_device{}());
    uniform_int_distribution<int> dist(100, 900);

    while (arr1.size() < n) {
        int num = dist(gen);
        if (num % 2 == 0) {
            arr1.push_back(num);
        }
    }
    return arr1;
}

// Генерация массива 2 из n/2 чисел в диапазоне [2, 50]
vector<int> GenerateArray2(int n) {
    vector<int> arr2;
    minstd_rand gen(random_device{}());
    uniform_int_distribution<int> dist(2, 50);

    while (arr2.size() < n / 2) {
        arr2.push_back(dist(gen));
    }
    return arr2;
}


//задание 2 
pair<vector<int>, vector<int>> Select_num(vector<int>& even_num, vector<int>& div_num){
    int count = 0;
    vector<int> ev_on_not_ev;
    for (const auto& num : even_num){
        if (count % 2 != 0){
            ev_on_not_ev.push_back(num);
        }
        count += 1;
    }

    count = 0;
    vector<int> not_ev_on_ev;
    for (const auto& num : div_num){
        if (count % 2 == 0 && num % 2 != 0){
            not_ev_on_ev.push_back(num);
        }
        count += 1;
    }

    return make_pair(ev_on_not_ev, not_ev_on_ev);
}

// Задание 3

vector<int> Div_mass(vector<int>& even_num, vector<int>& div_num, int k) {
    vector<int> nums;
    random_device rd;
    minstd_rand gen(rd());

    // Проверяем, что массивы не пустые и k > 0
    if (even_num.empty() || div_num.empty() || k <= 0) {
        return nums;
    }

    while (nums.size() < k) {
        uniform_int_distribution<> dist1(0, even_num.size() - 1);
        uniform_int_distribution<> dist2(0, div_num.size() - 1);

        int a = even_num[dist1(gen)];
        int b = div_num[dist2(gen)];

        // Исключаем деление на ноль
        if (b == 0) continue;

        int elem = a / b;
        nums.push_back(elem);
    }
    return nums;
}

// Задание 4
vector<char> GenerateSmb(int k){ // 
    random_device rd;
    minstd_rand gen(rd());

    vector<char> symb;
    uniform_int_distribution<int> dist (32, 126);
    while(symb.size() < k){
        char symbol = static_cast<unsigned char>(dist(gen));
        symb.push_back(symbol);
    }

    return symb;
}

vector<char> Del_cap_let(vector<char>& symbols){
    vector<char> result;
    for (const auto& ch : symbols) {
        if (!isupper(static_cast<unsigned char>(ch))) {
            result.push_back(ch);
        }
    }
    return result;
}

// Задание 5
vector<int> GenerateAndFilterArray(int n) {
    vector<int> arr;
    minstd_rand gen(random_device{}());
    uniform_int_distribution<int> dist(0, 300);

    while (arr.size() <= n){
        arr.push_back(dist(gen));
    }

    sort(arr.begin(), arr.end());

    vector<int> filtered;
    for (int num : arr) {
        int digits[10] = {0};
        int tmp = num;
        while (tmp > 0) {
            digits[tmp % 10]++;
            tmp /= 10;
        }
        for (int d : digits) {
            if (d >= 2) {
                filtered.push_back(num);
                break;
            }
        }
    }
    return filtered;
}

File: mass/go/commands.go
package main

import (
	"math/rand"
	"sort"
	"time"
	"unicode"
)

// Задание 1: Генерация массива 1 из n четных чисел в диапазоне [100, 900]
func GenerateArray1(n int) []int {
	arr := make([]int, 0, n)
	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
	for len(arr) < n {
		num := rnd.Intn(801) + 100 // [100, 900]
		if num%2 == 0 {
			arr = append(arr, num)
		}
	}
	return arr
}

// Генерация массива 2 из n/2 чисел в диапазоне [2, 50]
func GenerateArray2(n int) []int {
	arr := make([]int, 0, n/2)
	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
	for len(arr) < n/2 {
		num := rnd.Intn(49) + 2 // [2, 50]
		arr = append(arr, num)
	}
	return arr
}

// Задание 2: Выбор четных на нечетных местах и нечетных на четных местах
func SelectNum(evenNum, divNum []int) ([]int, []int) {
	evOnNotEv := make([]int, 0)
	for i, num := range evenNum {
		if i%2 != 0 {
			evOnNotEv = append(evOnNotEv, num)
		}
	}
	notEvOnEv := make([]int, 0)
	for i, num := range divNum {
		if i%2 == 0 && num%2 != 0 {
			notEvOnEv = append(notEvOnEv, num)
		}
	}
	return evOnNotEv, notEvOnEv
}

// Задание 3: Массив из k чисел - результат целочисленного деления случайных элементов
func DivMass(evenNum, divNum []int, k int) []int {
	nums := make([]int, 0, k)
	if len(evenNum) == 0 || len(divNum) == 0 || k <= 0 {
		return nums
	}
	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
	for len(nums) < k {
		a := evenNum[rnd.Intn(len(evenNum))]
		b := divNum[rnd.Intn(len(divNum))]
		if b == 0 {
			continue
		}
		nums = append(nums, a/b)
	}
	return nums
}

// Задание 4: Генерация массива случайных символов
func GenerateSmb(k int) []rune {
	symb := make([]rune, 0, k)
	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
	for len(symb) < k {
		symbol := rune(rnd.Intn(95) + 32) // [32, 126] печатаемые ASCII
		symb = append(symb, symbol)
	}
	return symb
}

// Удаление заглавных букв из массива символов
func DelCapLet(symbols []rune) []rune {
	result := make([]rune, 0, len(symbols))
	for _, ch := range symbols {
		if !unicode.IsUpper(ch) {
			result = append(result, ch)
		}
	}
	return result
}

// Задание 5: Массив из [0, 300], сортировка, фильтрация по повторяющимся цифрам
func GenerateAndFilterArray(n int) []int {
	arr := make([]int, 0, n)
	rnd := rand.New(rand.NewSource(time.Now().UnixNano()))
	for len(arr) < n {
		arr = append(arr, rnd.Intn(301)) // [0, 300]
	}
	sort.Ints(arr)
	filtered := make([]int, 0)
	for _, num := range arr {
		digits := [10]int{}
		tmp := num
		for tmp > 0 {
			digits[tmp%10]++
			tmp /= 10
		}
		for _, d := range digits {
			if d >= 2 {
				filtered = append(filtered, num)
				break
			}
		}
	}
	return filtered
}
